incipyt/tools/git.py

```diff
@@ -49,23 +52,55 @@ def pre(self, workon):
         """
         commands.run(["git", "init", os.fspath(workon)])
 
-        project.environ["AUTHOR_EMAIL"] = project.EnvValue(
-            commands.run(["git", "-C", os.fspath(workon), "config", "user.email"])
-            .stdout.decode()
-            .strip(),
-            update=True,
+        git_credential_email = commands.run(
+            ["git", "-C", os.fspath(workon), "config", "user.email"], check=False
         )
-        project.environ["AUTHOR_NAME"] = project.EnvValue(
-            commands.run(["git", "-C", os.fspath(workon), "config", "user.name"])
-            .stdout.decode()
-            .strip(),
-            update=True,
+        if git_credential_email.returncode == 0:
+            project.environ["AUTHOR_EMAIL"] = project.EnvValue(
+                git_credential_email.stdout.decode().strip(),
+                update=True,
+            )
+        else:
+            self._set_git_credential_email = True
+
+        git_credential_name = commands.run(
+            ["git", "-C", os.fspath(workon), "config", "user.name"], check=False
         )
+        if git_credential_name.returncode == 0:
+            project.environ["AUTHOR_NAME"] = project.EnvValue(
+                git_credential_name.stdout.decode().strip(),
+                update=True,
+            )
+        else:
+            self._set_git_credential_name = True
 
     def post(self, workon):
         """Run `git add --all`.
 
         :param workon: Work-on folder.
         :type workon: :class:`pathlib.Path`
         """
+        if self._set_git_credential_email:
```

Comment:
ok now I understand.

So the commit message should be:

	feat(git): user.name defaults to AUTHOR_NAME

	Make sure there is no global git user.name or user.email, make sure
	`git config --global user.name` is empty, same of `user.email`. Create a
	project using git and setuptools: the wizard prompts you for an author
	name and email as it couldn't determine them via git. When the project
	is created, the author name and email are set inside `setup.cfg` but the
	local git configuration still lacks an user.name and a user.email.

	In this commit when no git user name or email is set globally and that
	one was provided to the wizard, it automatically update the config of
	the new git repository to set the user.name and the user.email to the
	AUTHOR_NAME and AUTHOR_EMAIL.

and there should be a unittest covering it.